internal/build/container_resolver.go

```diff
@@ -18,13 +23,41 @@ func NewContainerResolver(dcli DockerClient) *ContainerResolver {
 	return &ContainerResolver{dcli: dcli}
 }
 
-// containerIdForPod looks for the container ID associated with the pod.
-// Expects to find exactly one matching container -- if not, return error.
-// TODO: support multiple matching container IDs, i.e. restarting multiple containers per pod
-func (r *ContainerResolver) ContainerIDForPod(ctx context.Context, podName k8s.PodID) (k8s.ContainerID, error) {
+// containerIdForPod looks for the container ID associated with the pod and image ID
+func (r *ContainerResolver) ContainerIDForPod(ctx context.Context, podName k8s.PodID, image reference.NamedTagged) (k8s.ContainerID, error) {
```

Comment:
niiiiice! we may want to rename this family of funcs to indicate that we're matching on image as well? or not, meh!